lib/queue.rb

```diff
@@ -1,31 +1,48 @@
 class Queue
-
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(100)
+    @front = -1
+    @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @front == -1 && @back == -1
+      @front = 0
+      @back = 1
+    end
+    if @front == @back
+      raise ArgumentError, "QUEUE IS FULL. PLEASE STOP STUFFING IT."
+    end
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
+    if @store[@front].nil?
+      @front = (@front + 1) % @store.length
+    end
```

Comment:
Why are you moving front when you `enqueue` here?  I suppose this occurs when you add to an empty list, but then front and back are both 1.  I don't think you need to do that.  You can handle the situation with an if-else above.